### ISSUE TITLE
Feature request: Add commit hash to build table

### ISSUE THREAD

**Issue by reece** [rtfd/readthedocs.org#2774]
It would be very handy for the "Recent Builds" table to indicate the commit hash (or changeset id for hg, etc). Even better would be a link to the commit and the first line of the commit message.

**Comment by stsewd** [comment]
Adding all the hash would be a little long, for git we can truncate to 7 characters and add a link to the repo.
I don't know how this works with the others vcs.

**Comment by reece** [comment]
I agree the full hash is long and unnecessary. 7 prefix chars is fine. The point is to be able to link to a specific changeset. Thanks! 

(And Mercurial will also accept prefixes of the revision hash)
